airflow/models.py

```diff
@@ -711,126 +711,56 @@ def is_runnable(self, flag_upstream_failed=False):
         """
         return self.is_queueable(flag_upstream_failed) and not self.pool_full()
 
-    def are_dependents_done(self, main_session=None):
-        """
-        Checks whether the dependents of this task instance have all succeeded.
-        This is meant to be used by wait_for_downstream.
-
-        This is useful when you do not want to start processing the next
-        schedule of a task until the dependents are done. For instance,
-        if the task DROPs and recreates a table.
-        """
-        session = main_session or settings.Session()
-        task = self.task
-
-        if not task._downstream_list:
-            return True
-
-        downstream_task_ids = [t.task_id for t in task._downstream_list]
-        ti = session.query(func.count(TaskInstance.task_id)).filter(
-            TaskInstance.dag_id == self.dag_id,
-            TaskInstance.task_id.in_(downstream_task_ids),
-            TaskInstance.execution_date == self.execution_date,
-            TaskInstance.state == State.SUCCESS,
-        )
-        count = ti[0][0]
-        if not main_session:
-            session.commit()
-            session.close()
-        return count == len(task._downstream_list)
-
     def are_dependencies_met(
             self, main_session=None, flag_upstream_failed=False):
-        """
-        Returns a boolean on whether the upstream tasks are in a SUCCESS state
-        and considers depends_on_past and the previous run's state.
 
-        :param flag_upstream_failed: This is a hack to generate
-            the upstream_failed state creation while checking to see
-            whether the task instance is runnable. It was the shortest
-            path to add the feature
-        :type flag_upstream_failed: boolean
-        """
-        TI = TaskInstance
         TR = TriggerRule
 
-        # Using the session if passed as param
         session = main_session or settings.Session()
+
         task = self.task
```

Comment:
we could use `self.task` in the rest of the code instead of creating a variable, this is more concise and inline with the style of access to other members